vertica_python/vertica/cursor.py

```diff
@@ -58,8 +58,7 @@ def execute(self, operation, parameters=None):
 
         self.rowcount = 0
         if self.last_execution:
-            # ToDo: can just empty the message buffer in connection if easy to do
-            self.connection.reset_connection()
```

Comment:
>>1) Is there a reason for setting _message to None?

Nope, not really. But it just makes sure that we start with a clean slate as it may have been populated earlier.

>>An example would be: user runs a SELECT query --> iterate/process the results and break the loop in the middle (a reason could be the condition is too complex to be embedded in the query) --> he runs another query --> undefined behavior because there are data left from the earlier query.

[TOO LONG CONTENT]
I think this needs to be detected and handled because we always know that we can fire a query to vertica only when the last message that we received from vertica is "ReadyForQuery". Maybe, this is complex to implement but I think this is cleaner than opening a new connection for every query fired. 

v0.4 could have broken API compatibility but it shouldn't have broken the conceptual compatibility with respect to how users model their programs to work with the library. And no where in README, it said that I shouldn't expect my "commit in separate query"  to work post 0.4.

I might want to choose to rewrite application code to use a different method in favor of other but it is definitely cumbersome to change every single query to have a commit in it which IMO is not natural anyways.